Migration of order view page printing functionality and layout changes - part 2

matks opened the pull request:
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | As PR https://github.com/PrestaShop/PrestaShop/pull/16262/ stayed out-of-sync with develop for too long, I rebuilt it in this PR.  When clicked "print" page display its version ready for print. There are some layout changes when print is being initialized.
| Type?         | new feature
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | #15819 
| How to test?  | Try clicking print or ctrl + p while in this page **admin-dev/index.php/sell/orders/orders/1/view**. It should be similar to invision sketch of when page is printed. **Dont forget to rebuild assets**

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/16700)
<!-- Reviewable:end -->

BlackKerio committed:
After modifying a product by webservice, the product position in the categories changes, even if we will use current position.

matks committed:
Add Money unit test

matks committed:
Fix MoneyTest unit test

Co-Authored-By: GoT <PierreRambaud@users.noreply.github.com>

matks committed:
Apply code review feedbacks for MoneyTest

djbuch committed:
Performance optimisation

We will only use the first row of this query, so limit to 1 in order to avoid querying multiple rows, because the join with stock_available will add as many rows than there are product attributes.

gerdus committed:
fix

BlackKerio committed:
Update classes/Product.php

Co-Authored-By: Pablo Borowicz <pablo.borowicz@prestashop.com>

BlackKerio committed:
Avoids webservice to create duplicated customers with same email

Commit:
fix: prevent order controller address validation from validating all guest addresses during guest checkout, should only apply to customers

Commit:
fix: coding style update

Commit:
Remove new-theme assets

Commit:
Remove admin default assets

Commit:
Remove themes assets

Commit:
Remove classic theme assets

Commit:
Create make file to install prestashop and build assets with 'make install' command

Commit:
Build assets in travis

BlackKerio committed:
Product.php

After the product has been modified by a webservice, while we also use associations, position in the categories changes

matks committed:
Cherry-pick of branch migration/order-view-page/printing-page

Commit:
First commit for eslint

Commit:
First step for eslint with autofix

LouiseBonnard commented on src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/details.html.twig:
```diff
@@ -23,46 +23,90 @@
+{% endset %}
+
+{% set carriersTitle %}
+  {{ 'Carriers'|trans({}, 'Admin.Orderscustomers.Feature') }} ({{ orderForViewing.shipping.carriers|length }})
```
```suggestion
  {{ 'Carriers'|trans({}, 'Admin.Shipping.Feature') }} ({{ orderForViewing.shipping.carriers|length }})
```

Commit:
Fixing print template

matks committed:
Fixing print template - 2

matks committed:
Update wording for print order

Co-Authored-By: LouiseBonnard <32565890+LouiseBonnard@users.noreply.github.com>

matks commented:
@MatShir Thanks to @NeOMakinG the "print" feature should be good now 😄 can you take a look ?

MatShir commented:
Let's try it :smile:

MatShir commented:
From a design, perspective is good for me, nice job.
[Orders • demo.pdf](https://github.com/PrestaShop/PrestaShop/files/3932522/Orders.demo.pdf)
@TristanLDD wdyt ?

@matks I noticed when I tested that some tabs are missing if you look from the screen above that the tab carriers is missing. Also that the payment tab goes on the second page, is it because of the hooks ? 
It was tested with order 5 from the fixtures.

Commit:
Stop displaying address icons and actions buttons

Commit:
Hiding change currency on PDF

MatShir commented:
@TristanLDD building the asset is better
[Demo with mass product.pdf](https://github.com/PrestaShop/PrestaShop/files/3932774/Demo.with.mass.product.pdf)

@NeOMakinG for the adjustment.

Commit:
Add lint-fix npm command

jolelievre commented on admin-dev/themes/new-theme/template/controllers/orders/page_header_toolbar.tpl:
```diff
@@ -37,3 +38,12 @@
   {/if}
 {/block}
+
+<div class="header-toolbar d-print-none">
```
I don't understand this new block? It wasn't there before and it's not used by any layout so what's its purpose?

matks replied:
It was a trick used to obtain this display:
<img src="https://user-images.githubusercontent.com/3830050/69636793-aedade80-1057-11ea-85bd-554da34f8c45.png">
However I reverted this trick (see https://github.com/PrestaShop/PrestaShop/pull/16560) as it was creating side-effects (see https://github.com/PrestaShop/PrestaShop/issues/16549)

jolelievre commented on admin-dev/themes/new-theme/scss/pages/_orders_view.scss:
```diff
@@ -0,0 +1,72 @@
+
+.adminorders {
+  @media print {
+    .header-toolbar {
```
What's the use of this since all `.header-toolbar` seem to have the `d-print-none` class anyway

NeOMakinG replied:
Probably not, we see the toolbar if we don't do this, I'll investigate a bit :D

jolelievre replied:
it just seems redundant

NeOMakinG replied:
Ah no ok you didn't understand the purpose of that style, the header-toolbar doesnt have the d-print-none class because we need to see it on the first page to get the main title, but it is in `position: fixed` by default, we need to avoid this header bar to be seen on every print pages so we set it to `position: initial`

jolelievre replied:
But yet in `page_header_toolbar.tpl` it has `d-print-none`

jolelievre commented on src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/details.html.twig:
```diff
@@ -23,46 +23,90 @@
-    <div class="tab-pane fade show active" id="historyTabContent" role="tabpanel" aria-labelledby="historyTab">
-      {% include '@PrestaShop/Admin/Sell/Order/Order/Blocks/View/history.html.twig' %}
+    <div class="tab-pane d-print-block fade show active" id="historyTabContent" role="tabpanel" aria-labelledby="historyTab">
+      {% embed '@PrestaShop/Admin/Sell/Order/Order/Blocks/View/details_card.html.twig' %}
```
This changes the display of the tabs right? not just for print view but in general (since it includes a card)
Is it what is expected in the design?

jolelievre replied:
Unless the card is only visible for print, and in screen view it seems as if there is no change thanks to `card--details` ?

NeOMakinG replied:
It just force the card to use `display: block` in print view so it take the full width space

jolelievre replied:
So is the card really necessary to do that? If the aim is only to have a `display: block;` on tabs it could be managed by adding it on `.tab-pane` directly If it was as simple as that then we could also apply it for hooks tabs
But I have the feeling the card is used so that the tab title may be printed close to the tab content, just a guess though Although I get the idea what bothers me is that this solution is a bit complex and won't work for hooks tabs.. (and I hope it doesn't modify the screen display)

jolelievre commented on src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/view.html.twig:
```diff
@@ -32,34 +32,48 @@
-  <div class="row">
-    <div class="col-md-12">
-      {% include '@PrestaShop/Admin/Sell/Order/Order/Blocks/View/order_actions.html.twig' %}
+  <div id="order-view-page">
```
Be careful with these modifications as they will conflict with https://github.com/PrestaShop/PrestaShop/pull/16144 if it is migrated sooner
This bring the question about how content from hooks will be interpreted during print.. Although you can't guess what they will print
Same question with the tabs from hooks https://github.com/PrestaShop/PrestaShop/pull/16144/files#diff-1a42492ba7a3b1fc9477d3ad77b8edb5R70
Which probably won't integrate card to optimize print display

NeOMakinG replied:
Well it will show the content of every tabs as it is doing with payment, shipping etc.. Should we force the display of default tabs and only these ?

jolelievre replied:
But I guess the card are used in order to have the title right before the content?
If so then I guess the tabs from hooks will look like:
```
Title 1
Title 2
Content 1
Content 2
```

NeOMakinG commented on admin-dev/themes/new-theme/scss/components/_button.scss:
```diff
@@ -1,6 +1,6 @@
 .btn {
   &.btn-action {
-    color: #576C72;
+    color: black;
```
```suggestion
    color: #000;
```

Respecting bootstrap standards

Commit:
Correcting review on header toolbar and colors

Commit:
Deleting non-used header-toolbar

Commit:
Revert "Deleting non-used header-toolbar" because header was used in print view

This reverts commit 7491c0c564abf7432573cbf7f501845a659fca2f.

Commit:
Deleting non-used header-toolbar

matks commented:
@jolelievre @NeOMakinG It seems there are 2 types of questions without answers here:
- what is the `page_header_toolbar.tpl` file and is it useful
- should we display, in the "print" display, the data/widgets/tabs/blocks added by modules ?

I'll answer the 1st question by going deeper in the code because I know this topic. IMO I
it was a trick used to obtain this display:
<img src="https://user-images.githubusercontent.com/3830050/69636793-aedade80-1057-11ea-85bd-554da34f8c45.png">
However I reverted this trick (see https://github.com/PrestaShop/PrestaShop/pull/16560) as it was creating side-effects (see https://github.com/PrestaShop/PrestaShop/issues/16549). I think we can delete it but I need to check.

For 2nd question I would suggest NOT to show modules additional content as indeed it could break the display. Or maybe provide an option to choose whether or not to display them. However this is a question that needs to be answered by @MatShir

matks commented:
> * what is the `page_header_toolbar.tpl` file and is it useful
> * should we display, in the "print" display, the data/widgets/tabs/blocks added by modules ?
> 
> I'll answer the 1st question by going deeper in the code because I know this topic.

I confirm it's useless now. It was deleted by Valentin so we're OK on this.

Commit:
Fix eslint errors

Commit:
Continue to fix eslint errors

Commit:
Implement eslint-vue

Commit:
Implement eslint-vue

Commit:
VueJs lint \o/

Commit:
Terminate with VueJs lint

Commit:
Finish the implementation of eslint for vue instances

Commit:
Use alias for scss files

Commit:
Build assets

Commit:
Implement the github action for eslint

BlackKerio committed:
Update classes/Customer.php

Co-Authored-By: Pablo Borowicz <pablo.borowicz@prestashop.com>

BlackKerio committed:
Update Customer.php

Commit:
Render template instead of using annotation

LouiseBonnard committed:
Add Bangladesh localization pack

LouiseBonnard committed:
Add Bosnia and Herzegovina localization pack

LouiseBonnard committed:
Add Cameroon localization pack

LouiseBonnard committed:
Add Senegal localization pack

LouiseBonnard committed:
Add Madagascar localization pack

Commit:
Fixing submenu not showing on classic theme

Commit:
Changing filters alignments and spaces

Commit:
Changing filters alignments and spaces

colinegin commented:
Needs to be merged in the end after all other PRs.

matks committed:
Remove useless customer query filter statements

Commit:
Update UI kit form theme to use Symfony bootstrap 4 theme as base

Commit:
Split UI kit form theme in two: normal layout (aka "base") and horizontal layout

Commit:
Allow setting up a custom class to a form row

Commit:
Form help now depends on each form_widget instead of on the form row

Commit:
Help is now rendered in raw

Commit:
Update Advanced Parameters > Performance to use the new form theme

Commit:
Adjusting quantity on product page if empty or under min qty

Commit:
CS fix

tdavidsonas88 committed:
Behat AddRootCategoryHandler

Commit:
Correcting history problem on listing page

tdavidsonas88 committed:
Behat DeleteCategoryCoverImageHandler

tdavidsonas88 committed:
Behat fixes for EditRootCategory test

Commit:
Remove duplicate var

tdavidsonas88 committed:
Behat DeleteCategoryCoverImage

tdavidsonas88 committed:
cover image variable no longer needed

tdavidsonas88 committed:
Behat DeleteCategoryMenuThumbnailImage

tdavidsonas88 committed:
Behat DeleteCategoryMenuThumbnailImage missing test case

tdavidsonas88 committed:
Behat SetCategoryIsEnabled

tdavidsonas88 committed:
Behat BulkUpdateCategoriesStatusCommand

tdavidsonas88 committed:
Behat GetCategoryIsEnabledHandler

tdavidsonas88 committed:
fixes in code style

tdavidsonas88 committed:
commented out file uploads

Commit:
Sending event on product refresh

tdavidsonas88 committed:
refactored the code after Tomas code review

Commit:
Fix eslint

Commit:
Fix eslint

tdavidsonas88 committed:
code style

Commit:
Add return type

Commit:
Remove default tranlsation_domain and use choice_translation_domain as needed

Commit:
Cs fix

Commit:
Github workflows

jolelievre committed:
Display symbols in form in case of error, remove DefaultLanguage constraint

Commit:
Apply comments

Commit:
Change global to window

Commit:
Kick element from callback as he's not used

Co-Authored-By: GoT <PierreRambaud@users.noreply.github.com>

jf-viguier committed:
email subject doesn't support html entities

Tools::safeOutput html entities encode subject that produce a buggy subject with special caracters.
The email subject is an smtp header ans doesn't support html.

fixes https://github.com/PrestaShop/PrestaShop/issues/16847

nesrineabdmouleh committed:
Crud address

nesrineabdmouleh committed:
Fix selectors name

Commit:
Fix lint

Commit:
New rule for the global global variable :)

LouiseBonnard committed:
Update the number of PrestaShop stores

nesrineabdmouleh committed:
Add some fixes

Commit:
Make optional fields optional again

Commit:
Fix history on product

Progi1984 committed:
Merge pull request #16961 from matks/remove-useless-customer-date-filters

Remove useless customer query filter statements

Commit:
Build assets

Progi1984 committed:
Merge pull request #17100 from PrestaShop/LouiseBonnard-patch-1

Update the number of PrestaShop stores

Commit:
Fix eslint

Progi1984 committed:
Merge pull request #16964 from eternoendless/update-performance

Refactor UI kit form theme and apply it to the Performance page

Progi1984 committed:
Merge pull request #16956 from NeOMakinG/issue14376

Changing classic theme filters alignments and spaces

Commit:
Fix disable shop function

SimonGrn committed:
Merge pull request #17150 from boubkerbribri/fixMaintenanceTest

Functional tests - Fix disable shop function

SimonGrn committed:
Revert "Functional tests - Fix disable shop function"

Progi1984 committed:
Merge pull request #17152 from PrestaShop/revert-17150-fixMaintenanceTest

Revert "Functional tests - Fix disable shop function"

matks committed:
Apply php-cs-fixer

matks committed:
Merge pull request #15431 from matks/add-money-test

Add Money unit test

tdavidsonas88 committed:
ps_tmp_img_dir inject through services

tdavidsonas88 committed:
fixed category injection bugg

tdavidsonas88 committed:
fixes for image creation

tdavidsonas88 committed:
possible better ways to solve upload issues - not sure if injection of services has any value now / in the future

tdavidsonas88 committed:
full beauty of the parameters names

tdavidsonas88 committed:
getting temp image dir from the container in test

SimonGrn committed:
Merge pull request #17086 from nesrineabdmouleh/functionalCRUDAddress

Functional tests - Crud customer address

Commit:
Merge branch '1.7.7.x' of github.com:PrestaShop/PrestaShop into merge2dev

jolelievre committed:
Merge pull request #17184 from PierreRambaud/merge2dev

Merge 1.7.7.x into develop - 15/01/2020

tdavidsonas88 committed:
refactored paths of images - chosen simpler solution

matks committed:
Merge branch '1.7.7.x' of github.com:PrestaShop/PrestaShop into merge2develop-5

matks committed:
Rebuild assets

PierreRambaud committed:
Merge pull request #17192 from matks/merge2develop-5

Merge 1.7.7.x into develop - 15/01/2020 - again

matks committed:
Add override hooks into Presenters

matks committed:
Add missing deprecation on previous PerformanceFormHandler

PierreRambaud committed:
Merge pull request #17204 from matks/add-deprecate-perf-form-handler-2

Add missing deprecation on previous PerformanceFormHandler

SimonGrn committed:
Revert "Functional tests - Crud customer address"

SimonGrn committed:
Merge pull request #17214 from PrestaShop/revert-17086-functionalCRUDAddress

Revert "Functional tests - Crud customer address"

matks committed:
Merge branch '1.7.7.x' of github.com:PrestaShop/PrestaShop into merge2develop-6

matks committed:
Merge pull request #17221 from matks/merge2develop-6

Merge 1.7.7.x into develop - 17/01/2020

Hherreros committed:
[FIX] Error when generating link_rewrite with API

When you create a product, you try to use the link_rewrite field as an array, but it has no value. This causes an error and the product doesnt add to DB. The array is initialized when is empty to fix this issue.

Hherreros committed:
Fix blank line

Fix a Blank line to pass Pretty CI Test

Commit:
Changing .closest to form attribute so the form can be outside

Commit:
Prettier quotes

tdavidsonas88 committed:
PHP_UNIT_FRAMEWORK imported - was missing import after rebase

tdavidsonas88 committed:
using Assert instead of PHPUnit_Framework_Assert.

matks committed:
Clean a bit files in install folder

matks committed:
Fix some links with http and no https

matthieu-rolland committed:
Merge pull request #17255 from matks/clean-install

Clean a bit files in install folder

tdavidsonas88 committed:
rolled back to $this->configuration->get('_PS_CAT_IMG_DIR_') in handler

Progi1984 committed:
Merge pull request #16965 from NeOMakinG/issue10125

Adjusting quantity on product page if empty or under min qty

Progi1984 committed:
Merge pull request #16927 from LouiseBonnard/develop

Add BA, BD, CM, MG and SN localization packs

PierreRambaud committed:
Merge pull request #17257 from matks/fix-https

Fix some links with http and no https

matks committed:
Add new badges to README

Progi1984 committed:
Merge pull request #17043 from jolelievre/fix-new-alternative-currency

Display symbols in form in case of error, remove DefaultLanguage constraint

matks committed:
Merge pull request #17284 from matks/add-badges

Add new badges to README

idnovate committed:
Fix https://github.com/PrestaShop/PrestaShop/issues/17290

PierreRambaud committed:
Merge pull request #16934 from NeOMakinG/issue16932

Fixing submenu not showing on classic theme

jolelievre committed:
Merge pull request #16670 from PierreRambaud/remove-useless-assets

Remove useless assets

matks committed:
Adapt test for negative amount for money

matks committed:
Merge branch '1.7.7.x' of github.com:PrestaShop/PrestaShop into merge2develop-8

matks committed:
Apply php-cs-fixer

matks committed:
Remove asset, they are not committed anymore on develop

jolelievre committed:
Merge pull request #17304 from matks/merge2develop-8

Merge 1.7.7.x into develop - 22/01/2020

tdavidsonas88 committed:
code style fix

Commit:
Build assets for nightly

Commit:
Remove github action to use PrettyCI only

SimonGrn committed:
Merge pull request #17319 from PierreRambaud/fix/nightly-assets

Build assets for nightly

matks committed:
Merge pull request #17018 from tdavidsonas88/feature/category-part-2=integration-tests

Cover Category CommandHandlers and QueryHandlers by behat tests  - Part 2

matks committed:
Merge pull request #17328 from PierreRambaud/fix/prettyci

Remove PHP GitHub Actions and use PrettyCI only

Commit:
Remove share buttons after install

Commit:
Add make assets while creating release

matks committed:
Merge branch '1.7.7.x' of github.com:PrestaShop/PrestaShop into merge2-develop-9

BlackKerio committed:
Merge branch 'develop' into patch-4

BlackKerio committed:
Update Product.php

BlackKerio committed:
Merge branch 'develop' into fix-webservice-duplicate-email

Progi1984 committed:
Merge pull request #16697 from BlackKerio/patch-4

Fix setWsPositionInCategory and setWsCategories, the product position in the categories changes

Commit:
Prevent order controller from validating all guest addresses (#16620)

Prevent order controller from validating all guest addresses

Progi1984 committed:
Merge pull request #17330 from PierreRambaud/fix/update-build-process

Add make assets while creating release

matks committed:
Merge pull request #17334 from matks/merge2-develop-9

Merge 1.7.7.x into develop - 23/01/2020

PierreRambaud committed:
Merge pull request #17340 from eternoendless/remove-installer-share-buttons

Remove share buttons after install

PierreRambaud committed:
Update PULL_REQUEST_TEMPLATE.md

matks committed:
Merge pull request #17369 from PrestaShop/update-github-template

Update PULL_REQUEST_TEMPLATE.md

Commit:
Remove unwanted git action

idnovate committed:
Update cart-summary-products.tpl

idnovate committed:
Update themes/classic/templates/checkout/_partials/cart-summary-products.tpl

Co-Authored-By: GoT <PierreRambaud@users.noreply.github.com>

matks committed:
Merge pull request #16553 from BlackKerio/fix-webservice-duplicate-email

Avoids webservice to create duplicated customers with same email

matks committed:
Merge pull request #16431 from gerdus/webservice-category-position-fix

Webservice: Updating category position is changed

matks committed:
Merge branch '1.7.7.x' of github.com:PrestaShop/PrestaShop into merge2develop-9

matks committed:
Remove monitoring asset from develop

PierreRambaud committed:
Do not force the npm to be updated in the build.sh file

matks committed:
Merge pull request #16704 from PierreRambaud/implement-eslint

Implement eslint on new-theme directory

matks committed:
Merge pull request #17387 from matks/merge2develop-9

Merge 1.7.7.x into develop - 28/01/2020

matks committed:
Fix code according to linter

jolelievre committed:
Merge pull request #17389 from PierreRambaud/fix/assets

Do not force the npm to be updated in the build.sh file

matks committed:
Merge pull request #17391 from matks/fix-develop-lint

Fix develop according to linter

micka-fdz committed:
Fix reversed product sorting

Progi1984 committed:
Merge pull request #17233 from Hherreros/develop

Fixed error when generating link_rewrite with API

Progi1984 committed:
Merge pull request #17402 from micka-fdz/reversed-product-sorting

Fix reversed product sorting

matks committed:
Enable closure to be appended into LazyArray

matks committed:
Apply php-cs-fixer

Commit:
Update eslint config to check for hidden directories

matks committed:
Fix lint on Product page

matks committed:
Improve Presenters hook naming

matks committed:
Fix mispelled property in LazyArrayImplementation

Progi1984 committed:
Merge pull request #17446 from matks/travis-youre-drunk

Fix PHP CS FIXER - again

Commit:
Remove assets

matks committed:
Merge pull request #17419 from matks/enable-closure-for-lazy-arrays

Enable closure to be appended into LazyArray

matks committed:
Merge pull request #17434 from PierreRambaud/update/eslint-directories

Update eslint config to check for hidden directories and files

matks committed:
Merge branch '1.7.7.x' of github.com:PrestaShop/PrestaShop into merge2develop-11

matks committed:
Apply lint to new-theme JS

matks committed:
Remove public assets

Commit:
Removing assets

matks committed:
Merge pull request #17466 from matks/merge2develop-11

Merge 1.7.7.x into develop - 31/01/2020

matks committed:
Merge pull request #17203 from matks/add-hooks-in-presenters

Add override hooks into Presenters

matks committed:
Merge pull request #17294 from idnovate/develop

Refresh cart summary when adding a voucher

Commit:
Let to const

matks committed:
Register new 1.7.8 hooks into XML and SQL files

matks committed:
Merge branch '1.7.7.x' of github.com:PrestaShop/PrestaShop into merge2develop-12

matks committed:
Fix hooks insert 1.7.8.0 request

matks committed:
Merge pull request #17483 from matks/merge2develop-12

Merge 1.7.7.x into develop - 03/02/2020

matks committed:
Merge pull request #17073 from jf-viguier/patch-6

Email subject doesn't support html entities

Matt75 committed:
Fix phpdoc of class Link

Commit:
Increase timeout to download translation packages from 5 to 20 seconds

PierreRambaud committed:
Merge pull request #17515 from eternoendless/increase-timeout

Increase timeout to download translation packages from 5 to 20 seconds

PierreRambaud committed:
Merge pull request #17510 from Matt75/phpdoc-link-core

Fix phpdoc of class Link

Progi1984 committed:
Merge pull request #15773 from djbuch/patch-57

CO : Performance optimisation

matks committed:
Only compute assets for Travis for tests which need it

Progi1984 committed:
Merge pull request #17133 from NeOMakinG/pierre/update-history

Correct history on product page

matks committed:
Use utf8mb4 in 1.7.8.0.sql upgrade script

Co-Authored-By: GoT <PierreRambaud@users.noreply.github.com>

Commit:
Merge branch 'develop' into 17194

Commit:
Resolve conflicts

Commit:
Fix lint errors

Commit:
Removing assets

matks committed:
Merge pull request #17020 from NeOMakinG/issue15184

Sending rating event on product refresh to avoid empty rating stars

Commit:
Correct tinymce conflict

Commit:
Remove unused JS in php comment

Commit:
Lint fi

matks committed:
Merge pull request #17545 from NeOMakinG/merge-into-develop

Merge 1.7.7.x into develop - 07/02/2020

matks committed:
Merge pull request #17014 from NeOMakinG/issue14821

Correct history problem on listing page

matthieu-rolland committed:
Merge pull request #17533 from matks/optimize-travis

Only compute assets for Travis for tests which need it

matks committed:
Merge pull request #17243 from NeOMakinG/17194

Permit submit form button to be outside of a form with a form attribute

matks committed:
Merge branch '1.7.7.x' of github.com:PrestaShop/PrestaShop into merge2develop-14

matks committed:
Merge pull request #17581 from matks/merge2develop-14

Merge 1.7.7.x into develop - 10/02/2020

matks committed:
Merge pull request #17477 from matks/add-new-hooks-presenters-data

Register new 1.7.8 hooks into XML and SQL files

zalexki committed:
update configuration::get phpdoc

PierreRambaud committed:
Merge pull request #17589 from zalexki/patch-3

Update Configuration::get phpdoc

Commit:
Merge branch 'develop' into migration/order-view-page/printing-page-2

Commit:
Revert "Merge branch 'develop' into migration/order-view-page/printing-page-2"

This reverts commit 2121292c3100531bcda7892f4b21337516315ac8, reversing
changes made to fcb3775d035d06af0f93705ba4d0724b0fc9b5e7.

PierreRambaud commented:
@matks You are trying to merge develop changes into the 1.7.7.x branch :sob:

NeOMakinG commented:
> @matks You are trying to merge develop changes into the 1.7.7.x branch 😭

No it's me and I reverted

matks commented:
PR rebuilt in https://github.com/PrestaShop/PrestaShop/pull/17617/ as we could not rebase it (too old)
